### ISSUE TITLE
Support for query parameters

### ISSUE THREAD

**Issue by tleunen** [ruyadorno/snapstub#10]
Currently, it seems the system doesn't support multiple api endpoints with different query parameters.

```
snapstub add http://example.com/api/v1/something/?q=query1
snapstub add http://example.com/api/v1/something/?q=query2
```

In this case, both api have different response, but only 1 response is saved under `__mocks`. The second one will overwrite the first one.

**Comment by ruyadorno** [comment]
hi @tleunen it would be def great to support that!

Ideally a system that can also support any variation of endpoints would be nice, since we also support POST data. I would probably have to leverage some other capacities from **stubborn** such as having an express middleware instead of a simple json file in order to parse requests and respond appropriately.

**Comment by zeachco** [comment]
@ruyadorno I don't know if snapstub supports the namespacing (sorry haven't played much with it yet)
but if it does, an alias could be created matching the parameters so the middleware could be very simple and that should work well with the current version of stuborn without any mutation. Alternatively, using the express middleware hook works as well I'm sure

**Comment by ruyadorno** [comment]
Landed on [v3.1.0](https://github.com/ruyadorno/snapstub/releases/tag/v3.1.0) 🎉
